src/modules/iKinGazeCtrl/src/controller.cpp

```diff
@@ -837,7 +837,8 @@ void Controller::run()
         // stabilization
         if (commData->stabilizationOn)
         {
-            Vector gyro=CTRL_DEG2RAD*commData->get_imu().subVector(6,8);
+            Vector gyro{commData->get_gyro().first};
+            reliableGyro = commData->get_gyro().second;
```

Comment:
Two things:
- you're calling here `get_gyro()` twice, hence two readings, unless the compiler will do some optimizations; wouldn't it be better to use a temp variable to avoid that?
- `reliableGyro` shouldn't be put equal to `false` only because of the network, as I suppose it'll be done by `getThreeAxisGyroscopeMeasure`; instead, it is declared `false` when a specific condition is met (seek for it in the code). Then, how will these operations interact with the intended strategy? Could you elaborate more on this?

Reply:
About the variable you are probably right, it is slightly better than do two different function calls.

> reliableGyro shouldn't be put equal to false only because of the network.

Giving a second glance maybe this part of code should be changed, probably all the computations and checks should be skipped if the MAS client doesn't receive any data, do you agree?

Reply:
- 👍  for 1.
- I agree with you that it'd be better to rethink 2, taking into account the original switch-off condition (that was unrelated to the network status).